### PR TITLE
Remove explicit transactions

### PR DESCRIPTION
**Pull request by apkar**
Explicit transactions are disabled for a while. We have a bunch of issues
related to connection affinity. They work only with sidecar setup. Even
then, they are generating quite a few corner cases with schema
management requests.

We have decided to go with MongoDB compatible transactions based on
sessions. Removing this code to make way for them.

### CODE REVIEW

File: test/correctness/document-correctness.py [hunk 1]
```diff
@@ -296,14 +297,18 @@ def _run_operation_(op1, op2):
...
             func1(*args1, **kwargs1)
         except pymongo.errors.OperationFailure as e:
+            print "Failed func1 with " + str(e)
             exceptionOne = e
         except MongoModelException as e:
+            print "Failed func1 with " + str(e)
             exceptionOne = e
         try:
-            func1(*args2, **kwargs2)
+            func2(*args2, **kwargs2)
```
Comment (dongxinEric) [inline test/correctness/document-correctness.py]: It was a luck/unfortunate that when using `transactional_shim`, `func1` and `func2` are always the same LOL. My bad anyways.

File: test/correctness/mongo_model.py [hunk 2]
```diff
@@ -495,7 +495,7 @@ def insert_one(self, dict):
     def insert_many(self, list):
         self.insert(list)
 
-    def find(self, query, fields=None):
+    def find(self, query, fields=None, batch_size=None):
```
Comment (dongxinEric) [inline test/correctness/mongo_model.py]: What's this `batch_size` used for?

Comment (apkar) [reply test/correctness/mongo_model.py]: `batch_size` from driver converts to `numberToReturn` in `OP_QUERY`. This is kind of page size for large queries.
